### ISSUE TITLE
@types/express-http-proxy filter function should return Promise<boolean or any>

### ISSUE THREAD

**Issue by ignaciocaff** [DefinitelyTyped/DefinitelyTyped#44550]
- [x] I tried using the `@types/express-http-proxy` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors:  @lrichb  @Daniel Schopf @Gabriel Fournier  @Niek van Bennekom @John L. Singleton  @Lindsay Wardell

Hi, filter inside httpProxy should return a Promise<boolean>, in order to allow make an async function to get permissions for example.

basically in index.d.ts 

change this ` filter?: (req: Request, res: Response) => boolean;`
for  `filter?: (req: Request, res: Response) => Promise<boolean>;`

I did it locally, and all works fine.

Greetings, Ignacio
